install_ddev.sh

```diff
@@ -14,26 +12,111 @@ BINOWNER=$(ls -ld /usr/local/bin | awk '{print $3}')
+#  -Do a git pull to ensure you have latest commits for core
+#  -Pre-load docker images for the sprint toolkit:
+#    -Drupal 8
+#    -phpmyadmin
+#    -Coud9 IDE
+#    -Thelounge IRC client
+#
+# Press y to continue
+# !!You don't need to hit enter!!.
+####
+${RESET}"
+read -n1 INSTALL
+if [[ ! $INSTALL =~ ^[Yy]$ ]]
+then
+    exit 1
+fi
+
+clear
 
 if [[ "$OS" == "Darwin" ]]; then
     FILEBASE="ddev_macos"
+
+    if ! command -v docker >/dev/null 2>&1; then
+        printf "
+        ${GREEN}
+        ####
+        # Installing docker so that ddev will work.
+        #
+        # Press y to continue
+        # !!You don't need to hit enter!!.
+        #
+        ####
+        ${RESET}"
+        read -n1 SEVEN
+        if [[ ! $SEVEN =~ ^[Yy]$ ]]
+        then
+            exit 1
+        fi
+
+        # Install and open Docker
+        hdiutil attach -nobrowse "${CURRENT_DIR}/docker_installs/Docker.dmg"
+        sleep 10
+        cp -rp /Volumes/Docker/Docker.app /Applications/
+        wait
+        open -a /Applications/Docker.app
+        hdiutil detach /Volumes/Docker
+
+        printf "
+        ${GREEN}
+        ####
+        # Please open Docker preferences and set Memory to 3.0 GiB on the Advanced tab.
+        # Wait for Docker to restart before continuing.
+        #
+        # Press y once this is done.
+        # !!You don't need to hit enter!!.
+        #
+        ####
+        ${RESET}"
+        read -n1 DOCKMEM
+        if [[ ! $DOCKMEM =~ ^[Yy]$ ]]
+        then
+            exit 1
+        fi
+    fi
+
 elif [[ "$OS" == "Linux" ]]; then
     FILEBASE="ddev_linux"
+
+    if ! docker --version >/dev/null 2>&1; then
+        printf "${YELLOW}Docker is required for ddev. Download and install docker at https://www.docker.com/community-edition#/download before attempting to use ddev.${RESET}\n"
+        printf "${YELLOW}See the Docker CE section at this page for linux installation instructions https://docs.docker.com/install/#server${RESET}\n"
+    fi
+
+    if ! docker-compose --version >/dev/null 2>&1; then
+        printf "${YELLOW}Docker Compose is required for ddev. Download and install docker-compose at https://www.docker.com/community-edition#/download before attempting to use ddev.${RESET}\n"
+        printf "${YELLOW}See the Docker CE section at this page for linux installation instructions https://docs.docker.com/install/#server${RESET}\n"
+    fi
+    
 else
-    printf "${RED}Sorry, this installer does not support your platform at this time.${RESET}\n"
+    printf "${RED}Sorry, this installer does not support your platform at this time.${RESET}\n"    
     exit 1
 fi
 
-if ! docker --version >/dev/null 2>&1; then
-    printf "${YELLOW}Docker is required for ddev. Download and install docker at https://www.docker.com/community-edition#/download before attempting to use ddev.${RESET}\n"
-fi
+echo "Installing docker images for ddev to use..."
+gzip -dc $(ls ddev_tarballs/ddev_docker_images*.tar.xz) | docker load
```

Comment:
I think I only commented in Slack on this, but this  and the similar line below also need a MacOS/Linux condition because gzip has native xz functionality in MacOS these days, but Linux does not. 

xzcat, similar to zcat, should extract xz to stdout.

`xzcat $(ls ddev_tarballs/ddev_docker_images*.tar.xz) | docker load`
`xzcat $(ls ddev_tarballs/docker_additions.tar.xz) | docker load`

Reply:
@mradcliffe do you know about it's existense and installation status on common dists?

Reply:
I have xzcat on Slackware 14.1/14.2 which is usually behind the times so I assume it's available on more hip distros like Ubuntu/Debian and Fedora/RHEL/CentOS.

Reply:
Try `docker run -it debian xzcat -h`, `docker run -it ubuntu xzcat -h`, `docker run -it centos xzcat -h` etc.